builtin/logical/pki/cert_util.go

```diff
@@ -255,16 +608,261 @@ func createCertificate(creationInfo *certCreationBundle) (*certutil.ParsedCertBu
 	if creationInfo.Usage&codeSigningUsage != 0 {
 		certTemplate.ExtKeyUsage = append(certTemplate.ExtKeyUsage, x509.ExtKeyUsageCodeSigning)
 	}
+	if creationInfo.Usage&emailProtectionUsage != 0 {
+		certTemplate.ExtKeyUsage = append(certTemplate.ExtKeyUsage, x509.ExtKeyUsageEmailProtection)
+	}
+
+	certTemplate.IssuingCertificateURL = creationInfo.URLs.IssuingCertificates
+	certTemplate.CRLDistributionPoints = creationInfo.URLs.CRLDistributionPoints
+	certTemplate.OCSPServer = creationInfo.URLs.OCSPServers
+
+	var certBytes []byte
+	if creationInfo.SigningBundle != nil {
```

Comment:
Is it possible to make this method assuming always having a signing bundle and then have a code path in the caller to deal with the self-signed case to turn it into the signing bundle? Just to reduce the complexity of this method